spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/AbstractDependsOnBeanFactoryPostProcessorTests.java

```diff
@@ -41,6 +41,14 @@
 	private ApplicationContextRunner contextRunner = new ApplicationContextRunner()
 			.withUserConfiguration(FooBarConfiguration.class);
 
+	@Test
+	void throwNoSuchBeanDefinitionSingletonBean() {
```

Comment:
Is that test failing without this change?

Reply:
Programmatically registered singleton does not have a `BeanDefinition`, hence test will fail independently of implementation.